src/typing/typer.ml

```diff
@@ -1852,7 +1852,28 @@ and type_expr ?(mode=MGet) ctx (e,p) (with_type:WithType.t) =
 		let e1 = vr#as_var "tmp" {e1 with etype = ctx.t.tnull e1.etype} in
 		let e_null = Builder.make_null e1.etype e1.epos in
 		let e_cond = mk (TBinop(OpNotEq,e1,e_null)) ctx.t.tbool e1.epos in
-		let iftype = WithType.WithType(e2.etype,None) in
+		let rec is_dead_end e = match e.eexpr with
+			| TParenthesis e -> is_dead_end e
+			| TThrow _ -> true
+			| TReturn _ -> true
+			| TBreak -> true
+			| TContinue -> true
+			| TWhile (_, body, DoWhile) -> is_dead_end body
+			| TIf (_, if_body, Some else_body) -> is_dead_end if_body && is_dead_end else_body
+			| TBlock exprs -> List.exists is_dead_end exprs
+			| TMeta (_, e) -> is_dead_end e
+			| TCast (e, _) -> is_dead_end e
+			| _ -> false in
```

Comment:
I know it looks silly, but for correctness we should consider compounds like `TCall` here because there could be `call(break)` or something along those lines.

Reply:
For that matter you should also check the function itself, e.g. `(break)(args)`.

Reply:
Oh god... but you're right.

This also affects several other compound expressions, like `TArray` and `TBinop` and such. Let's move this function to `tFunctions.ml` or somewhere and get it right.